client/state/plugins/marketplace/reducer.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import type { AnyAction } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { PLUGINS_MARKETPLACE_PRIMARY_DOMAIN_UPDATE } from 'calypso/state/action-types';
+import { combineReducers } from 'calypso/state/utils';
+
+export interface IPurchaseFlowState {
+	primaryDomain: string | undefined;
+}
+
+const defaultState: IPurchaseFlowState = {
+	primaryDomain: undefined,
+};
+
+function purchaseFlow(
+	state: IPurchaseFlowState = defaultState,
+	action: AnyAction
+): IPurchaseFlowState {
+	switch ( action.type ) {
+		case PLUGINS_MARKETPLACE_PRIMARY_DOMAIN_UPDATE:
+			return {
+				...state,
+				primaryDomain: action.domainName,
+			};
+		default:
+			return state;
+	}
+}
+
+export default combineReducers( {
```

Comment:
I'm assuming you're expecting this state tree to grow and that's why you've used `combineReducers` to create another level of hierarchy?

Generally, I'm noticing that this this tree seems to be pretty deep for the purpose: `state.plugins.marketplace.purchaseFlow.primaryDomain` - do we even need the `purchaseFlow` level? How do you envision the tree to look when this is feature-complete?

Reply:
Yes the vision for this part of the state tree is to have a home for all marketplace related state. In the future we might have a separate marketplace store similar to current plugins home. So any state required to manage that page will live here. Purchase flow will definitely grow in the future once we have multiple purchase flows and we try to make it more generic like the signup flow.